lookout/style/typos/metrics.py

```diff
@@ -0,0 +1,141 @@
+from typing import Dict, List, Set, Tuple
+
+import pandas
+
+from lookout.style.typos.utils import Columns
+
+
+class Scores:
+    """
+    Class to store scores of solutions of binary classification problems.
+    """
+
+    def __init__(self, tp: int = 0, fp: int = 0, tn: int = 0, fn: int = 0) -> None:
+        """
+        Initialize score counters.
+
+        :param tp: Number of true positive examples.
+        :param fp: Number of false positive examples.
+        :param tn: Number of true negative examples.
+        :param fn: Number of false negative examples.
+        """
+        self.tp = tp
+        self.fp = fp
+        self.tn = tn
+        self.fn = fn
+
+    def total(self) -> int:
+        """Get total number of examples."""
+        return self.tp + self.fp + self.tn + self.fn
+
+    def accuracy(self) -> float:
+        """Calculate accuracy."""
+        return (self.tp + self.tn) / self.total()
+
+    def precision(self) -> float:
+        """Calculate precision."""
+        return self.tp / (self.tp + self.fp)
+
+    def recall(self) -> float:
+        """Calculate recall."""
+        return self.tp / (self.tp + self.fn)
+
+    def f1(self) -> float:
+        """Calculate f1 score."""
+        return 2 / (1 / self.precision() + 1 / self.recall())
+
+    def get_metrics(self) -> Dict[str, float]:
+        """
+        Return accuracy, precision, recall and f1 scores.
+
+        :return: Dictionary `{metric_name: score}`.
+        """
+        return {"accuracy": self.accuracy(),
+                "precision": self.precision(),
+                "recall": self.recall(),
+                "f1": self.f1()}
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
+
+def first_k_set(corrections: List[Tuple[str, float]], k: int) -> Set[str]:
+    """
+    Compose the set of k most probable correction candidates (tokens without probabilities).
+
+    :param corrections: List of corrections, sorted by probability.
+    :param k: Number of corrections to take.
+    :return: Set of k most probable correction tokens.
+    """
+    first_k = set()
+    for correction, _prob in corrections[:k]:
+        first_k.add(correction)
+    return first_k
+
+
+def get_score(data: pandas.DataFrame, suggestions: Dict[int, List[Tuple[str, float]]],
```

Comment:
I wonder if we can re-use sklearn's functions here. `sklearn.metrics.classification_report`